corsikaio/subblocks/dtypes.py

```diff
@@ -2,22 +2,27 @@
 import numpy as np
 
 
-Field = namedtuple('Field', ['position', 'name', 'shape', 'dtype'])
+Field = namedtuple("Field", ["position", "name", "unit", "shape", "dtype"])
 
-# set defaults for shape and dtype, this works for the 2 right most attributes
-Field.__new__.__defaults__ = (1, 'f4')
+# set defaults for shape and dtype, this works for the 3 right most attributes
+Field.__new__.__defaults__ = (None, 1, "f4")
 
 
 def build_dtype(fields, itemsize=4 * 273):
     dt = defaultdict(list)
-    dt['itemsize'] = itemsize
+    dt["itemsize"] = itemsize
 
     for field in fields:
-        dt['names'].append(field.name)
-        dt['offsets'].append((field.position - 1) * 4)
+        dt["names"].append(field.name)
+        dt["offsets"].append((field.position - 1) * 4)
         if field.shape != 1:
-            dt['formats'].append((field.dtype, field.shape))
+            dt["formats"].append((field.dtype, field.shape))
         else:
-            dt['formats'].append(field.dtype)
+            dt["formats"].append(field.dtype)
 
     return np.dtype(dict(**dt))
+
+
+def build_data_dtype(fields):
```

Comment:
Why is that needed? Shouldn't the `build_dtype` function work for all lists of fields?

Reply:
Which means I think, that `build_dtype(fields, itemsize=None)` shoudl work for that!

Reply:
Yes, if you add do
```python
if itemsize is not None:
    dt["itemsize"] = itemsize
```

the second function is not needed.

Reply:
I did this because I notices that the resulting structured arrays didn't have also `offsets` and `formats` so I wanted to make sure I was replicating the dtype exactly.

should I also extend the condition to those 2 attributes?

Reply:
No, the defaults are fine for those.